kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -488,83 +494,102 @@ define([
          * Places it into the dom node
          * Hooks up event listeners
          */
-        function render() {
-            return Promise.try(() => {
-                const inputControl = makeInputControl(),
-                    content = div({ class: 'input-group', style: { width: '100%' } }, inputControl);
-
-                ui.setContent('input-container', content);
-                const dropdown = $(ui.getElement('input-container.input'));
-                const data = [];
-                if (config.isViewOnly) {
-                    let viewData = {
-                        selected: true,
-                        id: config.initialValue || undefined,
-                        text: config.initialValue,
-                    };
-                    if (config.initialDisplayValue) {
-                        if (typeof config.initialDisplayValue === 'string') {
-                            viewData.text = config.initialDisplayValue;
-                        } else if (
-                            config.initialDisplayValue !== null &&
-                            typeof config.initialDisplayValue === 'object'
-                        ) {
-                            viewData = Object.assign(viewData, config.initialDisplayValue);
-                            if (!viewData.text) {
-                                viewData.text = viewData.id;
-                            }
+        async function render() {
```

Comment:
Turned this function into `async` which tweaked the formatting. I'll point out the actual changes below.